extensions/Worker.Extensions.ServiceBus/src/ServiceBusSessionMessageActions.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.ServiceBus;
+using Google.Protobuf;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.Azure.ServiceBus.Grpc;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    [InputConverter(typeof(ServiceBusSessionMessageActionsConverter))]
+    public class ServiceBusSessionMessageActions
+    {
+        private readonly Settlement.SettlementClient _settlement;
+
+        private readonly string _sessionId;
+
+        internal ServiceBusSessionMessageActions(Settlement.SettlementClient settlement, string sessionId, DateTimeOffset sessionLockedUntil)
+        {
+            _settlement = settlement ?? throw new ArgumentNullException(nameof(settlement));
+            _sessionId = sessionId ?? throw new ArgumentNullException(nameof(sessionId));
+            SessionLockedUntil = sessionLockedUntil;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ServiceBusMessageActions"/> class for mocking use in testing.
+        /// </summary>
+        /// <remarks>
+        /// This constructor exists only to support mocking. When used, class state is not fully initialized, and
+        /// will not function correctly; virtual members are meant to be mocked.
+        ///</remarks>
+        protected ServiceBusSessionMessageActions()
+        {
+            _settlement = null!; // not expected to be used during mocking.
+        }
+
+        public virtual DateTimeOffset SessionLockedUntil { get; private set; }
+
+        ///<inheritdoc cref="ServiceBusReceiver.CompleteMessageAsync(ServiceBusReceivedMessage, CancellationToken)"/>
+        public virtual async Task<BinaryData> GetSessionStateAsync(
+            CancellationToken cancellationToken = default)
+        {
+            var request = new GetSessionStateRequest()
+            {
+                SessionId = _sessionId,
+            };
+
+            GetSessionStateResponse result = await _settlement.GetSessionStateAsync(request, cancellationToken: cancellationToken);
+            byte[] byteArray = result.SessionState.ToByteArray();
+            BinaryData binaryData = BinaryData.FromBytes(byteArray);
```

Comment:
We should be able to avoid copies/allocations here by creating the `BinaryData` instance over the `ReadonlyMemory<byte>` exposed by `ByteString`:

For example:
``` csharp
BinaryData binaryData = new BinaryData(result.SessionState.Memory);
```